feat: Enum: Add EnumConverter with underlying type lookup

riffy opened the pull request:
Add enum support. Only supports enum numbers

![grafik](https://github.com/duydang2311/AltV.Community.MValueAdapters.Generators/assets/13089522/4d7c398e-9cf9-4ab5-be44-4c59b04ac0ad)

riffy committed:
feat: Enum: Add EnumConverter with underlying type lookup

duydang2311 commented on AltV.Community.MValueAdapters.Generators/Converters/Other/EnumConverter.cs:
```diff
@@ -0,0 +1,45 @@
+{
+    protected override void GenerateItemWriteCode(StringBuilder stringBuilder, ref int indentation, MValueClassInfo classInfo, MValuePropertyInfo propertyInfo)
+    {
+        stringBuilder.AppendLine(indentation, $"writer.Value(({propertyInfo.UnderlyingEnumTypeName})value.{propertyInfo.Name});");
```
If the underlying type is one of the number types, the value should be casted one more time to the double type. It would be enum &rarr; underlying type &rarr; double, or simply enum &rarr; double.

riffy replied:
I've adapted the code to always make numbers (double) when generating `ToMValue` code.
It got a bit messy, so I cleaned up stuff.

Feel free to have a look

duydang2311 reviewed (changes requested):
Thanks for the PR!

Have you tested this between client and webview? I think it might not work because webview only considers double for number types, that was why if you have a look at numeric converters, they are all casted to double for that reason to keep the compatibility.

riffy committed:
feat: Enum: Make all numbers (double) on "ToMValue"
chore: Enum: Discrete differentiation between prefix and suffixes (makes it easier to extend later if necessary)

riffy commented:
> Thanks for the PR!
> 
> Have you tested this between client and webview? I think it might not work because webview only considers double for number types, that was why if you have a look at numeric converters, they are all casted to double for that reason to keep the compatibility.

I've not tested it with a webview, but since all numeric enums are now casted as doubles, this shouldn't be a problem

duydang2311 commented:
Actually, string/char enum isn't a real thing in C# so you can simply get rid of it for this PR. Can you update it so that it just has the underlying type only please? This looks good to be merged without the string/char thing.

I had an initial idea of supporting enum serialization as string that you would annotate your class/property with `[MValueMapEnum(EnumMappingStrategy.ByName)]`. The converter then uses `enumValue.ToString()` for write, and `Enum.Parse<YourEnum>(reader.NextString())` for read.

riffy committed:
Remove bool, string and char underlying enum type differnentiation

riffy commented:
I have removed bool, string and char. Only (double) casting is left.

duydang2311 commented:
> I have removed bool, string and char. Only (double) casting is left.
> 
> 

Perfect, thanks!
